python/iot-msk-lambda-pipeline/msk_demo/msk_demo_stack.py

```diff
@@ -0,0 +1,324 @@
+from constructs import Construct
+from aws_cdk import (
+    Duration,
+    NestedStack,
+    Stack,
+)
+from aws_cdk import (
+    aws_iot as iot,
+    aws_msk_alpha as msk,
+    aws_lambda as _lambda,
+    aws_ec2 as ec2,
+    aws_secretsmanager as sm,
+    aws_kms as kms,
+    aws_iam as iam
+)
+
+from aws_cdk.aws_lambda_event_sources import ManagedKafkaEventSource
+
+
+constants = {
+    "KAFKA_DOWNLOAD_VERSION": "kafka_2.12-2.6.2",
+    "KAFKA_VERSION": "2.6.2",
+    "KAFKA_CLIENT_INSTANCE": "t2.xlarge",
+    "MSK_TOPIC": "topic",
+    "IOT_TOPIC": 'iot/topic'
+}
+
+
+class LambdaConsumer(Stack):
+    def __init__(self, 
+                scope: Construct, 
+                construct_id: str,
+                cluster_arn,
+                vpc, 
+                msk_key: kms.IKey,
+                iot_task_policies, 
+                **kwargs):
+
+        super().__init__(scope, construct_id, **kwargs)
+        if not self.node.try_get_context("SecretFullArn"):
+            return
+
+        # Lambda function 
+        function = _lambda.Function(self, "LambdaFunction",
+                                    function_name="mskdemo_consume_message",
+                                    runtime=_lambda.Runtime.PYTHON_3_9,
+                                    vpc=vpc,
+                                    handler="lambda-handler.lambda_handler",
+                                    code=_lambda.Code.from_asset("./msk_demo/lambda"))
+
+        # Lambda function role 
+        function.role.attach_inline_policy(
+            iam.Policy(self, "LambdaRolePolicy",
+            statements=iot_task_policies)
+        )
+        
+        # MSK secret from full ARN (includes the 6-digit suffix) 
+        msk_secret = sm.Secret.from_secret_attributes(self, "MskSecret", 
```

Comment:
Why use fromSecretAttributes -- this implies the secret exists already.

CDK resources created for a stack should be created *in* a stack.

Reply:
The secret is being created by `msk_cluster.add_user('demo')`. I have not found a way to get the cluster's secret complete arn outside of this method.